service/host_mock.go

```diff
@@ -32,7 +32,12 @@ import (
 
 // NopHost is to be used in tests
 func NopHost() Host {
-	return NopHostConfigured(auth.NopClient, ulog.NopLogger, opentracing.NoopTracer{})
+	return NopHostWithConfig(nil)
+}
+
+// NopHostWithConfig is to be used in tests and allows setting of config.
+func NopHostWithConfig(configProvider config.Provider) Host {
```

Comment:
Could you replace a similar struct: https://github.com/uber-go/fx/blob/master/modules/rpc/thrift_test.go#L37 ?